ted2zim/templates/assets/article.js

```diff
@@ -0,0 +1,25 @@
+$.urlParam = function(name){
+    var results = new RegExp('[?&]' + name + '=([^&#]*)').exec(window.location.href);
+    if (results==null) {
+       return null;
+    }
+    return decodeURI(results[1]) || 0;
+}
+
+window.onload = function() {
+    html_name = window.location.pathname.substring(window.location.pathname.lastIndexOf("/") + 1);
+    video_id = html_name.substring(0, html_name.lastIndexOf("."));
+    for (i in json_data) {
+        if(json_data[i].id == video_id){
+            var idx = 0;
+            for (j in json_data[i].title) {
+                if (json_data[i].title[j].lang == $.urlParam('lang')) {
```

Comment:
you're calling this for each iteration of this nested loop. call it once and store it.

Reply:
Yup. That's definitely better. Did that.